### PR TITLE
Add public Do method to http client

### LINKED ISSUE #5: Response Headers
What if I need access to the response headers? Right now the Response [1] struct has no way to get them.

[1] https://github.com/gojektech/heimdall/blob/master/response.go#L4

### PR DESCRIPTION
**Pull request by sohamkamani**
Fixes #5 , #4 , #3 

- Make Do method public
- Fix data race condition for hystrix client

### CODE REVIEW

File: http_client.go [hunk 1]
```diff
@@ -110,50 +114,60 @@ func (c *httpClient) Delete(url string, headers http.Header) (Response, error) {
...
 
-	return c.do(request)
+	res, err := c.Do(request)
+	return toHeimdallResponse(res, err)
 }
 
-func (c *httpClient) do(request *http.Request) (Response, error) {
-	hr := Response{}
+func toHeimdallResponse(response *http.Response, err error) (Response, error) {
+	fmt.Println(response, err)
```
Comment (rShetty) [inline http_client.go]: We should remove this

File: http_client.go [hunk 2]
```diff
@@ -110,50 +114,60 @@ func (c *httpClient) Delete(url string, headers http.Header) (Response, error) {
...
 	request.Header = headers
 
-	return c.do(request)
+	res, err := c.Do(request)
+	return toHeimdallResponse(res, err)
 }
 
-func (c *httpClient) do(request *http.Request) (Response, error) {
-	hr := Response{}
+func toHeimdallResponse(response *http.Response, err error) (Response, error) {
```
Comment (rShetty) [inline http_client.go]: I think we should get rid of converting to heimdall Response and instead return *http.Response back

File: http_client.go [hunk 3]
```diff
@@ -110,50 +114,60 @@ func (c *httpClient) Delete(url string, headers http.Header) (Response, error) {
...
+		response.Body.Close()
+	}
+	hr.statusCode = response.StatusCode
+	return hr, merr.HasError()
+}
 
+// Do makes an HTTP request with the native `http.Do` interface
+func (c *httpClient) Do(request *http.Request) (*http.Response, error) {
 	request.Close = true
 	multiErr := valkyrie.NewMultiError()
```
Comment (rShetty) [inline http_client.go]: The valkyrie no longer supports constructor based initialization. We should change it.

File: http_client.go (hunk 1 above)
Comment (sohamkamani) [reply http_client.go]: 👍

File: http_client.go [hunk 4]
```diff
@@ -110,50 +114,60 @@ func (c *httpClient) Delete(url string, headers http.Header) (Response, error) {
...
-		response.Body.Close()
-
-		hr.statusCode = response.StatusCode
-
 		if response.StatusCode >= http.StatusInternalServerError {
 			multiErr.Push(fmt.Sprintf("server error: %d", response.StatusCode))
 
 			backoffTime := c.retrier.NextInterval(i)
 			time.Sleep(backoffTime)
+			fmt.Println("R: ", response.StatusCode)
```
Comment (rShetty) [inline http_client.go]: Need to remove this

**Comment by rShetty** [comment]
Closing this one, duplicate of #10

**Pull request closed without merging**